### DISCUSSION TITLE
Best Practices for Describing Keys?

### DISCUSSION THREAD

**Discussion by bmulholland** [envkey/envkey#30] [Q&A] [answered]
It's easy to lose track of the purpose of keys, and it's easy to keep around keys that are no longer needed. To combat that, our current credentials approach stores a descriptive comment alongside each secret. This appears not to be possible in EnvKey; I couldn't even see a way to add text like this as a commit message. What do other EnvKey users do to solve these problems? Where to document each key, and how to ensure it stays up to date? Does EnvKey have a feature like this planned?

**Comment by danenania** [comment, marked as answer]
Good question @bmulholland. A per-variable notes/comment feature is definitely on the roadmap and we have some groundwork laid for it. As with just about everything in EnvKey, including variable names, we want these notes to be end-to-end encrypted, so it is a bit more complex than it might sound. Certainly still doable though!  

While not particularly elegant, I suppose in the meantime you could add a `VARIABLE_COMMENT`  below each `VARIABLE` you want to document, or else keep a separate file for this purpose in source control.

**Comment by drcongo** [comment]
A vaguely related possible feature request here could be variable grouping / scoping - even if that's just a presentational nicety on top of the desktop UI. I'm thinking for example a group called `POSTGRES` with variables inside for `HOST`, `PASSWORD` etc. which would be accessed as `POSTGRES_HOST` and `POSTGRES_PASSWORD`. We've ended up with quite a lot of stuff stored in our EnvKey projects, only some of which is needed on the remote servers, so this could also enable the ability to group vars by why they're needed - ie a `CI` group. You'd probably still want the comments feature though, as that could give the context of each group too.

**Reply by danenania** [reply]
Hey @drcongo, could the [blocks](https://docs-v2.envkey.com/docs/blocks) feature possibly fit what you're looking for in terms of variable grouping?

**Reply by drcongo** [reply]
Not quite I don't think, as this looks like it would share the values across projects - this does look like a super useful feature that I had no idea existed though and would definitely help reduce the number of variables we have per project!

**Reply by danenania** [reply]
That's good to hear! You're right that it's mainly intended to share values across projects, but if you extract groups of vars into blocks and then include them in just a single project, it would in effect break those vars up into different sections in the UI as you're describing. It's true though that we could make grouping vars within a single app a bit smoother than this even if it uses blocks underneath.

**Reply by drcongo** [reply]
Ahh, got you. That's a pretty clever workaround, thanks!
